src/dbnode/storage/bootstrap/bootstrapper/persist.go

```diff
@@ -288,15 +292,17 @@ func BuildBootstrapIndexSegment(
 		return err
 	}
 
-	segments := []segment.Segment{NewSegment(seg, false)}
-	for _, seg := range indexBlock.Segments() {
-		segments = append(segments, seg)
-	}
+	for volumeType, indexBlock := range indexBlockByVolumeType.Data {
+		segments := []segment.Segment{NewSegment(seg, false)}
+		for _, seg := range indexBlock.Segments() {
+			segments = append(segments, seg)
+		}
 
-	// Now replace the active segment with the built segment.
-	newFulfilled := indexBlock.Fulfilled().Copy()
-	newFulfilled.AddRanges(expectedRanges)
-	replacedBlock := result.NewIndexBlock(blockStart, segments, newFulfilled)
-	indexResults[xtime.ToUnixNano(blockStart)] = replacedBlock
+		// Now replace the active segment with the built segment.
+		newFulfilled := indexBlock.Fulfilled().Copy()
+		newFulfilled.AddRanges(expectedRanges)
+		replacedBlock := result.NewIndexBlock(segments, newFulfilled)
```

Comment:
Is this the correct logic here? Looks like we're creating a single segment from the builder and then creating a block for each type of volume type (i.e. potentially two index blocks for the same segment). Same with the other method in this package `PersistBootstrapIndexSegment`. 

Maybe we should instead just return the replaced block and let the caller insert it into the index results for both of these functions?

Reply:
Either that or key the index results the opposite way, i.e. by volume type first, then by block start (so you can pass the second level depth map into this method and keep the code the way it was before).

Reply:
So it's still only creating a single `IndexBlockByVolumeType` per `blockStart`. `IndexBlock`s are added to the the `Data` property of that struct keyed by the `IndexVolumeType`.

The behaviour should be the same as before?

Reply:
Spoke offline. Decision is to change the persist fn method API to return an index block and allow the caller to figure out where to place it. 

In this case, it would replace the "default" index volume type block.